### ISSUE TITLE
Rule for detecting unneccessary braces aroung the trailing lambda

### ISSUE THREAD

**Issue by atulgpt** [pinterest/ktlint#1949]
## Expected Rule behaviour
The rule should flag that the trailing lambda can be used without the braces around it. Examples of compliant and non-compliant code

```
// non-compliant code
fun test() {
    repeat(10, {
       println(it)
    })
}

// compliant code
fun test() {
    repeat(10) {
        println(it)
    }
}
```

## Additional information
* Current version of ktlint: NA
* Styleguide section: NA

**Comment by paul-dingemans** [comment]
Unfortunately that is not possible to do with a Ktlint rule. Ktlint has no knowledge about the signature of function `repeat` and as of that can not determine whether the lambda is the last parameter in the function.

Given function `foo` below:
```
    fun foo(bar: Boolean, block: () -> String) =
        if (bar) {
            block()
        } else {
            "no bar"
        }
```
then `foo(true) { "Foo test" }` is a valid call.

However, given function `foo2` below:
```
    fun foo2(block: () -> String, bar: Boolean = true) =
        if (bar) {
            block()
        } else {
            "no bar"
        }
```
then `foo2(true) { "Foo test" }` or `foo2 { "Foo test" }` are not valid (compile error).

In order to check this, Ktlint should have more information about the call context which is out of scope for this project.

**Comment by atulgpt** [comment]
@paul-dingemans Thanks for the detailed explanation
